test/stale_state_test.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2022, Pulumi Corporation.  All rights reserved.
```

Comment:
Testing the flow of this feature. Here we create a RabbitMQ stack/container, and another stack that interacts with RabbitMQ. We then go through our expectations that it:

1. Works expected normally if nothing is stale
2. Fails if the container attempting to ping RabbitMQ has old credentials in State
3. Succeeds to interact with RabbitMQ if we create a new Stack CR to specifically refresh the stale secret
4. Succeeds if we define the new stack as a prerequisite refresh Stack, and the parent stack is enqued.